frontend/packages/shared/src/components/GiteaHeader/VersionControlButtons/components/ShareChangesPopover/CommitAndPushContent/CommitAndPushContent.tsx

```diff
@@ -4,26 +4,45 @@ import { useTranslation } from 'react-i18next';
 import { useVersionControlButtonsContext } from '../../../context';
 import { Heading, Paragraph } from '@digdir/design-system-react';
 import { StudioButton, StudioTextarea } from '@studio/components';
+import type { RepoContentStatus } from 'app-shared/types/RepoStatus';
+import { ClockDashedIcon } from '@studio/icons';
+import { FileChangesInfoModal } from './FileChangesInfoModal/FileChangesInfoModal';
 
 export type CommitAndPushContentProps = {
-  handleClosePopover: () => void;
+  onHidePopover: (hide: boolean) => void;
+  onClosePopover: () => void;
+  fileChanges: RepoContentStatus[];
 };
 
-export const CommitAndPushContent = ({ handleClosePopover }: CommitAndPushContentProps) => {
+export const CommitAndPushContent = ({
+  onHidePopover,
+  onClosePopover,
+  fileChanges,
+}: CommitAndPushContentProps) => {
   const { t } = useTranslation();
   const { commitAndPushChanges } = useVersionControlButtonsContext();
 
   const [commitMessage, setCommitMessage] = useState('');
+  const [showFileChangesIsOpen, setShowFileChangesIsOpen] = useState(false);
 
   const handleClickCommitAndPush = async () => {
     await commitAndPushChanges(commitMessage);
-    handleClosePopover();
+    onClosePopover();
   };
 
   const handleTextareaChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
     setCommitMessage(event.target.value);
   };
 
+  const handleModalOpen = () => {
+    setShowFileChangesIsOpen(true);
+    onHidePopover(true);
+  };
+  const handleModalClose = () => {
+    setShowFileChangesIsOpen(false);
+    onHidePopover(false);
+  };
```

Comment:
```suggestion
const handleModalToggle = (isOpen) => {
  setShowFileChangesIsOpen(isOpen);
  onHidePopover(isOpen);
};
```
What you think about combining these two functions into a single toggle function? Where the respective calls can decide if they should pass true or false

Reply:
Nice suggestion! Will change 🤩